features/step_definitions/administration_steps.rb

```diff
@@ -0,0 +1,118 @@
+Given(/^I have an admin account$/) do
+  FactoryGirl.create(:admin_user)
+end
+
+When(/^I visit the admin panel url$/) do
+  visit new_admin_user_session_path
+end
+
+When(/^I enter my admin email$/) do
+  fill_in 'Email', with: 'admin@example.com'
+end
+
+When(/^I enter my admin password$/) do
+  fill_in 'Password', with: 'password'
+end
+
+Then(/^I see the admin panel$/) do
+  visit admin_root_path
+end
+
+Given(/^I do not have an admin account$/) do
+end
+
+Then(/^I see a flash notification that tell me that my email does not exist in the system$/) do
+  expect(page).to have_content("Invalid email or password.")
+end
+
+Given(/^I am logged into the admin panel$/) do
+  FactoryGirl.create(:admin_user)
+  visit new_admin_user_session_path
+  fill_in 'Email', with: 'admin@example.com'
+  fill_in 'Password', with: 'password'
+  click_on "Login"
+end
+
+Given(/^I am logged into the site$/) do
+  FactoryGirl.create(:user, email: "ruti@test.com")
+  ctoken = ActionMailer::Base.deliveries.last.body.match(/confirmation_token=\w*/)
```

Comment:
Here I am confirming the token sent in my confirmation email. But is there a better way to do this? It passes the test suite 90% (I think) of the time, but fails sometimes I assume because the email hasn't fired. Any thoughts are welcome!